src/main/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/services/VersionService.kt

```diff
@@ -0,0 +1,75 @@
+package uk.gov.justice.digital.hmpps.sentenceplan.services
+
+import jakarta.persistence.EntityManager
+import jakarta.persistence.NoResultException
+import jakarta.persistence.PersistenceContext
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.sentenceplan.entity.AreaOfNeedEntity
+import uk.gov.justice.digital.hmpps.sentenceplan.entity.PlanVersionEntity
+import uk.gov.justice.digital.hmpps.sentenceplan.entity.PlanVersionRepository
+import uk.gov.justice.digital.hmpps.sentenceplan.entity.StepEntity
+import java.util.UUID
+
+@Service
+class VersionService(
+  private val planVersionRepository: PlanVersionRepository,
+) {
+  @PersistenceContext
+  lateinit var entityManager: EntityManager
+
+  /**
+   * This function makes a copy of a PlanVersion object and all its descendent objects.
+   * New UUIDs are set on the copied objects and then persisted to the database.
+   */
+  @Transactional
+  fun createNewPlanVersion(planVersionUuid: UUID): PlanVersionEntity {
+    val newPlanVersionEntity: PlanVersionEntity
+
+    try {
+      newPlanVersionEntity = planVersionRepository.getWholePlanVersionByUuid(planVersionUuid)
+    } catch (e: NoResultException) {
```

Comment:
Can use `_` as `e` is never used.

Reply:
Resolved in https://github.com/ministryofjustice/hmpps-sentence-plan/pull/104/commits/35c138c3305c49a6e5495fbdba9018d327b2cba8